src/java/cheshire/prettyprint/CustomPrettyPrinter.java

```diff
@@ -0,0 +1,87 @@
+package cheshire.prettyprint;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+
+import java.io.IOException;
+
+class DynamicIndenter extends DefaultIndenter {
+    public DynamicIndenter(int indentation) {
+        super(new String(new char[indentation]).replace("\0", " "), "\n");
```

Comment:
Instead of specifying indentation as number of characters in the Java file, I'd recommend making this take a String, then you could have someone decide they want indentation with `\t` instead of spaces if they want.

Then, from the Clojure side you can convert the number `2` into `"  "` and not have to do it here